src/checkPublication.ts

```diff
@@ -1,93 +1,171 @@
-            imageCount: result.Contents.filter(object => object.Key.endsWith('.jpg')).length
+            articleCount: result.Contents.filter(object =>
+                object.Key.endsWith(".nitf.xml")
+            ).length,
+            imageCount: result.Contents.filter(object => object.Key.endsWith(".jpg"))
+                .length
         };
 
         if (info.articleCount >= config.MinimumArticleCount) {
-            return Promise.resolve(info)
+            return Promise.resolve(info);
         } else {
-            return Promise.reject(`Expected at least ${config.MinimumArticleCount} articles, but there are only ${info.articleCount}`)
+            return Promise.reject(
+                `Expected at least ${
+                config.MinimumArticleCount
+                } articles, but there are only ${info.articleCount}`
+            );
         }
     };
 
-    let sendEmail = (subject: string, body: string, targetAddresses: string[]): Promise<SendEmailResponse> => {
+    const getLogs = (logGroupName: string): Promise<Array<AWS.CloudWatchLogs.OutputLogEvent>> => findLogStreams(logGroupName).then(getCloudwatchLogs(logGroupName));
+
+    const findLogStreams = (logGroupName: string): Promise<Array<AWS.CloudWatchLogs.LogStream>> => new Promise((resolve, reject) => {
+        logs.describeLogStreams({
+            logGroupName,
+            orderBy: 'LastEventTime',
+            descending: true,
+            limit: 2
+        }, function (err, data) {
+            if (err) {
+                reject(err);
+                return;
+            }
+            resolve(data.logStreams);
+        })
+    });
+
+    const getCloudwatchLogs = (logGroupName: string) => (logStreams: AWS.CloudWatchLogs.LogStream[]): Promise<Array<AWS.CloudWatchLogs.OutputLogEvent>> => {
+        const getLogsFor = (logStream: AWS.CloudWatchLogs.LogStream): Promise<AWS.CloudWatchLogs.OutputLogEvent[]> => new Promise((resolve, reject) => {
+            logs.getLogEvents({
+                logGroupName,
+                logStreamName: logStream.logStreamName,
+                startFromHead: true
+            }, (err, data) => {
+                if (err) {
+                    reject(err);
+                    return;
+                } else if (data.events.length < 20) {
+                    throw new Error(`Log stream ${logStream.logStreamName} in ${logGroupName} only contains ${data.events.length} but we expected hundreds of them`)
+                }
+
+                resolve(data.events);
+            })
+        });
+        return getLogsFor(logStreams[0]).catch(() => getLogsFor(logStreams[1]));
+    }
+
+    const checkLogs = (allLogs: Array<AWS.CloudWatchLogs.OutputLogEvent>): Promise<void> => {
+        const errors = allLogs.filter(log => / WARN | ERROR | FATAL /.test(log.message))
+        if (errors.length > 0) {
+            return Promise.reject(errors)
+        } else {
+            return Promise.resolve();
+        }
+    }
```

Comment:
another one (`;`) - probably not essential these days but consistency FTW! And on line 144 I think 🙄